src/System.Private.ServiceModel/tests/Stress/SharedPoolsOfWCFObjects/CallStats.cs

```diff
@@ -0,0 +1,197 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Collections.Concurrent;
+using System.Threading;
+using System.Diagnostics;
+
+namespace WCFClientStressTests
+{
+    /// <summary>
+    /// This class provides a few Call*AndRecordStats() utility methods which will store and aggregate the timing of the delegates they invoke 
+    /// 
+    /// Note: all Call*AndRecordStats() methods catch all exceptions coming from the delegates
+    /// However we don't want to miss the case when an unexpected exception happens for the calls that should never have exceptions
+    /// </summary>
+    public class CallStats
+    {
+        public CallTimingStatsCollector SunnyDay { get; set; }
+        public CallTimingStatsCollector RainyDay { get; set; }
+        public CallStats(int samples = 1000000, int errorSamples = 10000)
+            : this(samples, errorSamples, CallTimingStatsCollector.CalculateSomePercentiles, CallTimingStatsCollector.CalculateSomePercentiles)
```

Comment:
These integer values should probably be turned into public static fields at the top file.  As we get further along (and if you can take a dependency on TestProperties), we can find a way to make them configurable from the command line.